amy/workshops/migrations/0229_membership_M2M_organisations.py

```diff
@@ -4,7 +4,12 @@
 import django.db.models.deletion
 
 
-DEFAULT_ROLES = (("default", "Default"),)
+DEFAULT_ROLES = (
```

Comment:
So I don't typically see folks editing migrations. It can cause headaches and creates dependency problems. Especially since anyone who has run the migration already will not get your changes. Could you tell me more about why you feel this is better than creating a new migration?

Reply:
In general it's okay to change this migration (`0229`) because it's only released in staging environment of AMY. I don't usually change existing migrations

In this specific case I had to revert migrations up to `0228` in the staging env because of the feedback I received (the result of the feedback is this PR).

Unfortunately during the rollback to `0228` it turned out we had issue with Django rendering model states in a way that prevented re-creating the `membership.organization` column. This forced me to change migration `0065` - it's benign change for the system (and not represented by code anymore), but helps Django correctly render model state when rolling back migration 0229.

However, changing `0065` forced me to change `0229` in `copy_organisation_to_organisations` - because the field was nullable I had to account for that.

Finally the reverse operation for `copy_organisation_to_organisations` - `copy_default_organisations_to_organisation` - had some small corrections I wanted to do (it had assumed there was only 1 Member per membership).


----

I think the biggest lessons learned for me here is to always test the rollback migration. I was kind-of okay with `0229` in the first place because I only anticipated changes in the default roles, however as it turned out reverting `0229` wasn't as simple as I had thought.

Reply:
Gotcha! That's helpful context. I understand. I'll keep that in mind moving forward. By the way how do you know what has been released to production. is that simply whatever is on main?

Reply:
There's a couple pointers:

1. `main` should always be the branch released in prod env. Sometimes I have to make a hotfix (literally a hot fix in prod) so there's some uncommitted code in prod, but I always bring it officially through a minor release from the `develop` branch
2. there are tags marking release versions (e.g. `v2.16`) made on `main` branch; AMY release is visible in the footer and should correspond to the release tag
3. I'm deploying the code so I know which tag is applied 😃